rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,453 @@
+  "typ": "JWT",
+  "x5c": [
+    "<base64 encoded leaf-certificate in the DER format>",
+    "<base64 encoded issuer-intermediate-certificate in the DER format>",
+    "<base64 encoded issuer-certificate in the DER format>"
+  ],
+  "x5t": "<thumbprint>",
+  "kid": "did:x509:0:sha256:<hash>::subject:O:Library%20The%20Bookworm::subject:L:Bookland::san:otherName:123#1"
+}
+```
+
+(TODO: Add right JSON-LD context)
+Payload:
+
+```json
+{
+  "vc": {
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1"
+    ],
+    "type": [
+      "VerifiableCredential",
+      "X509Credential"
+    ],
+    "issuer": "did:x509:0:sha256:<hash>::subject:O:Library%20The%20Bookworm::subject:L:Bookland::san:otherName:123",
+    "issuanceDate": "2024-12-01T00:00:00Z",
+    "credentialSubject": {
+      "id": "did:web:example.com",
+      "subject:O": "Library The Bookworm",
+      "subject:L": "Bookland",
+      "san:otherName": "123"
+    }
+  }
+}
+```
+
+## Validation
+
+To validate an `X509Credential`, the following steps MUST be performed:
+
+- Verify that the credential is in JWT format.
+- Verify that the issuer's DID is a `did:x509` DID.
+- Resolve the `did:x509` DID document according to
+  the [did:x509 specification](https://trustoverip.github.io/tswg-did-x509-method-specification/) and check the
+  certificate chain for revocation.
+- Validate that the `credentialSubject` fields match the policies in the `did:x509` DID.
+
+### 1. Verify Credential Structure
+
+Ensure that the credential:
+
+- Includes the `X509Credential` type.
+- Contains a valid `did:x509` issuer.
+- Includes a `credentialSubject` whose fields match the `did:x509` DID Document.
+
+### 2. Validate the Issuer Certificate
+
+The certificate associated with the `did:x509` issuer MUST be validated as follows:
+
+- **Certificate Chain Validation**: The certificate must have a valid trust chain. The use case determines if the CA is
+  trusted.
+- **Revocation Check**: Verify the revocation status of the certificate using OCSP or CRL.
+
+Failure to validate the issuer certificate invalidates the credential.
+
+### 3. Validate the Credential Subject
+
+The `credentialSubject` MUST be verified against the `did:x509` DID Document. Specifically:
+
+- Every field in the `credentialSubject` MUST be present in the `did:x509` DID Document.
+- Fields not present in the `did:x509` DID Document invalidate the credential.
+
+### 4. Verify the Proof
+
+The cryptographic proof of the credential MUST be verified using the public key associated with the `did:x509` DID.
+This involves:
+
+- Resolving the public key from the DID Document.
+- Verifying the signature on the credential's `proof`.
+
+### 5. Check Credential Expiry
+
+If the `issuanceDate` or any other relevant date constraints (e.g., `expirationDate`) are present, they MUST be
+validated to ensure the credential is within its valid timeframe.
+
+## Security Considerations
+
+The following security considerations are to be considered:
```

Comment:
also state which catastrophic failure might occur when you don't do this properly